src/KeyUnit.cpp

```diff
@@ -138,8 +138,7 @@ bool KeyUnit::disableKey(const QString& name)
 
 bool KeyUnit::killKey(QString& name)
 {
-    for (auto it = mKeyRootNodeList.begin(); it != mKeyRootNodeList.end(); it++) {
-        TKey* pChild = *it;
+    for (auto pChild : mKeyRootNodeList) {
```

Comment:
Trouble is that now it is not totally obvious (to us humans) what the `auto` type is resolved to by the compiler... 🤷‍♂️

Reply:
It is - just hover your mouse over it!

![workspace 1_317](https://user-images.githubusercontent.com/110988/40955605-5cdc1bb4-688b-11e8-8e21-049e2b511d19.png)

Reply:
No it doesn't (in my Qt 5.9.4 setup using Creator 4.5.1)... 😦 
![syntax_highlight_issue](https://user-images.githubusercontent.com/6163092/40992716-78409164-68ef-11e8-850b-5e85c6d97f2c.png)

Reply:
Perhaps you don't have the clang code model enabled? https://doc.qt.io/qtcreator/creator-clang-codemodel.html

Reply:
Humm, I can't see it, and it does not seem enable-able:
![no_clangcodemodel](https://user-images.githubusercontent.com/6163092/41075774-1866342a-6a06-11e8-8205-8383e6e371b9.png)
![no_clangcodemodel1](https://user-images.githubusercontent.com/6163092/41075786-202eff5c-6a06-11e8-9f88-68e894a01295.png)

Ah, I think I will have to wait for Qt Creator 4.6.1 to make it into 11.1-RELEASE: https://www.freshports.org/search.php?query=qtcreator

As the Qt Wiki seems to imply that 4.6 is needed : https://wiki.qt.io/Qt_Creator_Clang_Code_Model

Ironically - I have to use the clang compiler system as it is the FreeBSD default and the GCC compilers produce binaries that seem to crash during the initialisation of the CRT.

Reply:
Ah ok.